libraries/primitive_consul_lock.rb

```diff
@@ -72,6 +72,7 @@ def human_duration(duration_in_secs)
 
     def wait_until(action, opts = {})
       dc = "(in #{@options[:datacenter]})" if @options[:datacenter]
+      opts[:max_failures] ||= @options[:max_failures]
```

Comment:
You are updating the users' input, this is not super great in term of side effect. Could you store it in a local variable?